torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -568,8 +568,11 @@ void check_gpu_tensors(const std::vector<at::Tensor>& tensors) {
     if (t.sizes() != first.sizes()) {
       throw std::runtime_error("Tensors must have identical size");
     }
-    if (!t.is_contiguous()) {
```

Comment:
Does this mean all ops in `ProcessGroupNCCL` no longer require contiguous? Or does `is_non_overlapping_and_dense` implies contiguous?

Reply:
> Does this mean all ops in ProcessGroupNCCL no longer require contiguous? 
Correct.  They require all inputs are non overlapping and dense with the same layout.  It turns out DDP didn't actually need this feature, but I wasn't sure that would end up true until after I had finished the DDP changes.